crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -0,0 +1,1218 @@
+                )
+            },
+            || {
+                let bytes = ByteView::open(cfi_path)
+                    .map_err(|err| {
+                        let stderr: &dyn std::error::Error = &err;
+                        tracing::error!(stderr, "Error while reading cficache");
+                    })
+                    .ok()?;
+
+                let cfi_cache = CfiCache::from_bytes(bytes)
+                    // This mostly never happens since we already checked the files
+                    // after downloading and they would have been tagged with
+                    // CacheStatus::Malformed.
+                    .map_err(|err| {
+                        let stderr: &dyn std::error::Error = &err;
+                        tracing::error!(stderr, "Error while loading cficache");
+                    })
+                    .ok()?;
+
+                if cfi_cache.as_slice().is_empty() {
+                    return None;
+                }
+
+                SymbolFile::from_bytes(cfi_cache.as_slice())
+                    .map_err(|err| {
+                        let stderr: &dyn std::error::Error = &err;
+                        tracing::error!(stderr, "Error while processing cficache");
+                    })
+                    .ok()
+            },
+        )
+    }
+
+    fn missing_ids(self) -> Vec<DebugId> {
+        self.missing_ids.into_inner().into_iter().collect()
+    }
+}
+
+#[async_trait]
+impl SymbolProvider for SymbolicatorSymbolProvider {
+    async fn fill_symbol(
+        &self,
+        module: &(dyn Module + Sync),
+        _frame: &mut (dyn FrameSymbolizer + Send),
+    ) -> Result<(), FillSymbolError> {
+        let debug_id = module.debug_identifier().ok_or(FillSymbolError {})?;
+
+        // Symbolicator's CFI caches never store symbolication information. However, we could hook
+        // up symbolic here to fill frame info right away. This requires a larger refactor of
+        // minidump processing and the types, however.
+        match self.files.contains_key(&debug_id) {
+            true => Ok(()),
+            false => Err(FillSymbolError {}),
+        }
+    }
+
+    async fn walk_frame(
+        &self,
+        module: &(dyn Module + Sync),
+        walker: &mut (dyn FrameWalker + Send),
+    ) -> Option<()> {
+        let debug_id = module.debug_identifier()?;
+        match self.files.get(&debug_id) {
+            Some(file) => file.walk_frame(module, walker),
+            None => {
+                self.missing_ids.write().insert(debug_id);
+                None
+            }
+        }
+    }
+
+    fn stats(&self) -> HashMap<String, SymbolStats> {
+        self.files
+            .iter()
+            .map(|(debug_id, sym)| {
+                let stats = SymbolStats {
+                    symbol_url: sym.url.clone(), // TODO(ja): We could put our candidate URI here
+                    loaded_symbols: true, // TODO(ja): Should we return `false` for not found?
```

Comment:
since we're on a TODO-addressing spree, these might be worth taking a look at - @jan-auer might have more context on these if he remembers them though.